### PR TITLE
Add a tool to verify that script files stop upon seeing an error

### PR DESCRIPTION
**Pull request by duglin**
see: https://github.com/kubernetes-incubator/service-catalog/issues/317

Signed-off-by: Doug Davis <dug@us.ibm.com>

### CODE REVIEW

**Comment by duglin** [comment]
CI issues doesn't seem related.

File: verify/verify-errexit.sh [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  args="$*"
+else
+  args=$(ls "$REPO_ROOT" | grep -v vendor | grep -v glide)
+fi
+
+# Gather the list of files that appear to be shell scripts.
+# Meaning they have some form of "#!...sh" as a line in them.
+shFiles=$(grep -rl '^#!.*sh$' $args)
+
+tmp=/tmp/out$RANDOM
```
Comment (ixdy) [inline verify/verify-errexit.sh]: rather than using a tempory file, maybe just start with `rc=0`?

File: verify/verify-errexit.sh [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+for file in ${shFiles}; do
+  grep "set -o errexit" $file > /dev/null 2>&1 && continue
+  grep "set -[a-z]*e" $file > /dev/null 2>&1 && continue
+
+  echo $file: appears to be missing \"set -o errexit\" | tee -a $tmp
```
Comment (ixdy) [inline verify/verify-errexit.sh]: and then set `rc=1` here rather than writing to the tmpfile.

File: verify/verify-errexit.sh [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  args=$(ls "$REPO_ROOT" | grep -v vendor | grep -v glide)
+fi
+
+# Gather the list of files that appear to be shell scripts.
+# Meaning they have some form of "#!...sh" as a line in them.
+shFiles=$(grep -rl '^#!.*sh$' $args)
+
+tmp=/tmp/out$RANDOM
+for file in ${shFiles}; do
+  grep "set -o errexit" $file > /dev/null 2>&1 && continue
```
Comment (ixdy) [inline verify/verify-errexit.sh]: a shorter alternative to `> /dev/null 2>&1` is `&>/dev/null`

style nit: also please quote variables and use `${var}` form, so `"${file}"` here.

File: verify/verify-errexit.sh [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+# Usage: verify-errexit.sh [ dir | file ... ]
+# default args is the root of our source tree
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+REPO_ROOT=$(dirname "${BASH_SOURCE}")/..
+
+if [ "$*" != "" ]; then
```
Comment (ixdy) [inline verify/verify-errexit.sh]: style nit: we prefer `[[ ]]` over `[ ]`

File: verify/verify-errexit.sh [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+  args="$*"
+else
+  args=$(ls "$REPO_ROOT" | grep -v vendor | grep -v glide)
```
Comment (ixdy) [inline verify/verify-errexit.sh]: style nit: we prefer `${var}` over `$var`.

**Comment by ixdy** [comment]
I think if you rebase your PR travis will pass.

File: verify/verify-errexit.sh (hunk 5 above)
Comment (ixdy) [inline verify/verify-errexit.sh]: also, I suspect you may want to make the excludes list configurable at some point, possibly excluding specific scripts.

File: verify/verify-errexit.sh (hunk 4 above)
Comment (duglin) [inline verify/verify-errexit.sh]: done

File: verify/verify-errexit.sh (hunk 5 above)
Comment (duglin) [inline verify/verify-errexit.sh]: fixed nit. I'd like to do the configurable list in a follow-on PR if that's ok

File: verify/verify-errexit.sh (hunk 1 above)
Comment (duglin) [inline verify/verify-errexit.sh]: hmm no idea why I did this the hard way :-)  I think it was a copy-n-paste from a different script
fixed

File: verify/verify-errexit.sh (hunk 3 above)
Comment (duglin) [inline verify/verify-errexit.sh]: fixed

File: verify/verify-errexit.sh (hunk 2 above)
Comment (duglin) [inline verify/verify-errexit.sh]: done

**Comment by duglin** [comment]
sorry for the delay - fixed the nits and rebased

**Comment by duglin** [comment]
ping @ixdy

**Comment by ixdy** [comment]
lgtm, though it appears that the script is missing +x.

**Comment by duglin** [comment]
@ixdy good catch!  fixed the +x issue - think it should be all good to go now.

**Comment by duglin** [comment]
@ixdy thanks

**Comment by ixdy** [comment]
I realized after merging that we perhaps should have added this check to `.travis.yml` here. Oh well.

**Comment by duglin** [comment]
good point - I'll open another PR for that....

**Pull request merged**